src/components/calculator.js

```diff
@@ -1,69 +1,61 @@
-import React from 'react';
+/* eslint-disable react/destructuring-assignment */
 
+import React from 'react';
+import calculate from '../logic/calculate';
 import './calculator.css';
 
-const numbers = () => {
-  const array = [];
-  for (let i = 9; i >= 7; i -= 1) {
-    array.push(
-      <button type="button" id={i}>{i}</button>,
-    );
-  }
-  return array;
-};
-
-const number2 = () => {
-  const array = [];
-  for (let i = 6; i >= 4; i -= 1) {
-    array.push(
-      <button type="button" id={i}>{i}</button>,
-    );
+// eslint-disable-next-line react/prefer-stateless-function
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { total: null, next: null, operation: null };
+    this.handleEvent = this.handleEvent.bind(this);
   }
-  return array;
-};
 
-const number3 = () => {
-  const array = [];
-  for (let i = 1; i <= 3; i += 1) {
-    array.push(
-      <button type="button" id={i}>{i}</button>,
-    );
+  handleEvent(e) {
+    // eslint-disable-next-line react/no-access-state-in-setstate
+    this.setState(calculate(this.state, e.target.textContent));
   }
-  return array;
-};
 
-// };
-// eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
   render() {
     return (
       <div className="calculator">
         <header className="head">
-          <span>0</span>
+          <span>
+            {this.state.total}
+            {this.state.next}
+            {this.state.operation}
```

Comment:
[OPTIONAL]

- I think it will be better if you use object destructuring to get the state variables so that your code will be easier to read 👍🏽 

e.g

` const { total, next, operation } = this.state; `

Reply:
I will implement this as well. Thank you for the suggestions.